lib/puppet/network/http/connection.rb

```diff
@@ -200,6 +203,89 @@ def request_with_redirects(request, options)
       raise RedirectionLimitExceededException, _("Too many HTTP redirections for %{host}:%{port}") % { host: @host, port: @port }
     end
 
+    # Handles the Retry-After header of a HTTPResponse
+    #
+    # This method checks the response for a Retry-After header and handles
+    # it by sleeping for the indicated number of seconds. The response is
+    # returned unmodified if no Retry-After header is present.
+    #
+    # @param response [Net::HTTPResponse] A response recieved from the
+    #   HTTP client.
+    #
+    # @return [nil] Sleeps and returns nil if the response contained a
+    #   Retry-After header that indicated the request should be retried.
+    # @return [Net::HTTPResponse] Returns the `response` unmodified if
+    #   no Retry-After header was present or the Retry-After header could
+    #   not be parsed as an integer or RFC 2822 date.
+    def handle_retry_after(response)
+      retry_after = response['Retry-After']
+      return response if retry_after.nil?
+
+      retry_sleep = parse_retry_after_header(retry_after)
+      server_hostname = case response.uri
+                        when URI
+                          response.uri.host
+                        else
+                          # Net::HTTPResponse was generated by a method that
+                          # did not fill in the uri attribute, so use a
+                          # placeholder for the server name.
+                          _('the remote server')
```

Comment:
We're trying to clean up dangling standalone phrases like this that need to be translated, because they're easy to translate wrong (e.g. here you can't tell if this is the subject of the sentence or not, which will matter for some languages like German). But in this case I don't see a good way to refactor this to make a whole sentence without duplicating a bunch of code. Can you instead just add a `TRANSLATORS` comment here indicating how this phrase is used? Something like "used in the phrase 'Received a response from the remote server'".

Reply:
Thanks for the advice! Added a TRANSLATORS comment.

Reply:
Thanks 👍